ts/containers/staking/wizard/wizard.ts

```diff
@@ -28,10 +38,29 @@ const mapDispatchToProps = (dispatch: Dispatch<Action>): ConnectedDispatch => {
         onOpenConnectWalletDialog: (): void => {
             dispatcher.updateIsConnectWalletDialogOpen(true);
         },
+        onDepositAndStartStakingAsync: async (
+            providerState: ProviderState,
+            networkId: Network,
+            amountToStakeInput: string,
+        ): Promise<void> => {
+            const amountToStakeBaseUnits = Web3Wrapper.toBaseUnitAmount(
+                new BigNumber(amountToStakeInput, 10),
+                constants.DECIMAL_PLACES_ZRX,
+            );
+
+            const account = providerState.account as AccountReady;
+            const currentAllowance = account.zrxAllowanceBaseUnitAmount || new BigNumber(0);
+
+            if (amountToStakeBaseUnits.isGreaterThan(currentAllowance)) {
+                await asyncDispatcher.increaseZrxAllowanceAndDispatchToStoreIfNeededAsync(
+                    providerState,
+                    networkId,
+                    amountToStakeBaseUnits,
+                    dispatcher,
+                );
+            }
```

Comment:
Longer term you might want this code in the actual component just because we'll need to be awaiting in there to show the loading animations.

Reply:
Makes sense, I will iterate on this logic after finishing the upgrade 👍